kiss

```diff
@@ -443,16 +451,24 @@ pkg_source_get() {
 }
 
 pkg_source_git() {
-    # This magic will shallow clone branches, commits or the
-    # regular repository. It correctly handles cases where a
-    # shallow clone is not possible.
-    log "$repo_name" "Cloning $1"
-
     # Split the source into URL + OBJECT (branch or commit).
     url=$1 com=${url##*[@#]} com=${com#${url%[#@]*}}
 
-    git init
-    git remote add origin "${url%[#@]*}"
+    if [ -z "$(ls -A)" ]; then
```

Comment:
You could simply check to see if `.git` exists as a directory no?

Reply:
Replaced it with git rev-parse